manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -179,8 +190,15 @@ public Processor updateProcessor(String bridgeId, String processorId, String cus
         Set<BaseFilter> existingFilters = existingDefinition.getFilters();
         String existingTransformationTemplate = existingDefinition.getTransformationTemplate();
         Action existingAction = existingDefinition.getRequestedAction();
-        Source existingSource = existingDefinition.getRequestedSource();
         Action existingResolvedAction = existingDefinition.getResolvedAction();
+        Source existingSource = existingDefinition.getRequestedSource();
+
+        // for sensitive fields we may receive either a string with the new value or an empty object
+        // if it remains unchanged (since the caller can't know the current real value)
+        // thus, before comparing the requested action/source with the existing one, we must unmask
+        // the sensitive fields with the original values (if those are unchanged) or with the new ones
+        Action unmaskedRequestedAction = unmaskRequestedGateway(processorRequest.getAction(), existingAction);
+        Source unmaskedRequestedSource = unmaskRequestedGateway(processorRequest.getSource(), existingSource);
```

Comment:
Do you also need to update [these](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/fa09fc86ad2c9f6146b2ace88970d61415555709/manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java#L240-L241) lines to use the unmasked values?

Reply:
I just fixed the update :+1: